meeting.js

```diff
@@ -0,0 +1,330 @@
+    localVideo.srcObject = null;
+    document.getElementById("localvideoimage").style.display = "block";
+    videoButton.innerHTML = '<i class="fas fa-video-slash icon red"></i>';
+    isVideoStarted = false;
+    vbgButton.classList.remove('selected');
+    if (vbgEffect && vbgEffect.isEnabled) {
+      await vbgEffect.disable();
+    }
+  } 
+  // Start Video
+  else {
+    const cameraStream = await webex.meetings.mediaHelpers.createCameraStream({ width: 640, height: 480 });
+    localStream.camera = cameraStream;
+    localVideo.srcObject = cameraStream.outputStream;
+    await createdMeeting.publishStreams({ camera: localStream.camera });
+    videoButton.innerHTML = '<i class="fas fa-video icon green"></i>';
+    document.getElementById("localvideoimage").style.display = "none";
+    isVideoStarted = true;
+  }
+}
+
+// 13) Mute and UnMute Button using toggleMicrophone Function
+microphoneButton.addEventListener('click', toggleMicrophone);
+async function toggleMicrophone() {
+  if (!localStream) {
+    console.error('No local stream available.');
+    return;
+  }
+  // Unmute Local Audio
+  if (!isMuted) {
+    try {
+      const microphoneStream = await webex.meetings.mediaHelpers.createMicrophoneStream({
+        echoCancellation: true,
+        noiseSuppression: true,
+      });
+
+      localStream.microphone = microphoneStream;
+      await createdMeeting.publishStreams({ microphone: localStream.microphone });
+      microphoneButton.innerHTML = '<i class="fas fa-microphone icon green"></i>';
+      microphoneButton.style.padding = "14px 21px";
+      isMuted = true;
+    } catch (error) {
+      console.error('Error creating microphone stream:', error);
+    }
+  } 
+  // Mute Local Audio
+  else {
+    if (localStream.microphone) {
+      await createdMeeting.unpublishStreams([localStream.microphone]);
+      localStream.microphone = null;
+      bnrButton.classList.remove('selected');
+    }
+    microphoneButton.innerHTML = '<i class="fas fa-microphone-slash icon red"></i>';
+    microphoneButton.style.padding = "14px 17px";
+    isMuted = false;
+  }
+}
+
+// 16) Added the Local Share System using Share Button.
+shareButton.addEventListener('click', localshare);
+async function localshare() {
+  const [localShareVideoStream, localShareAudioStream] =
+    await webex.meetings.mediaHelpers.createDisplayStreamWithAudio();
+  await createdMeeting.publishStreams({
+    screenShare: {
+      video: localShareVideoStream,
+      audio: localShareAudioStream,
+    }
+  });
+}
+
+// 17) Enabled BNR using Event Listeners and toggleBNR Function.
+bnrButton.addEventListener('click', toggleBNR);
+async function toggleBNR() {
+  let bnrEffect = null;
+  if (!localStream || !localStream.microphone) {
+    console.error('No local microphone stream available.');
+    return;
+  }
+
+  try {
+    if (!bnrEffect) {
+      const audioContext = new AudioContext({ sampleRate: 48000 });

```

Comment:
Creating the audio context is optional. We do not need that. Let's create noise reduction effect without a custom audio context and let the WxEffects library handle it

Reply:
Updated